ui/shared/components/panel/view-fields/BaseFieldView.jsx

```diff
@@ -18,76 +20,126 @@ const FieldValue = styled.div`
   display: ${props => ((props.fieldName && !props.compact) ? 'block' : 'inline-block')};
 `
 
-const BaseFieldView = (props) => {
-  if (props.isVisible !== undefined && !props.isVisible) {
-    return null
+class BaseFieldView extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      showInLineButton: true,
+    }
+    this.toggleButtonVisibility = this.toggleButtonVisibility.bind(this)
   }
-  if (props.isPrivate && !props.user.isStaff) {
-    return null
-  }
-  const fieldValue = props.initialValues[props.field]
-  const hasValue = (fieldValue && (!Object.getOwnPropertyNames(fieldValue).includes('length') || fieldValue.length > 0)) || props.showEmptyValues
-  if (!props.isEditable && !hasValue) {
-    return null
+
+  toggleButtonVisibility() {
+    this.setState({
+      showInLineButton: !this.state.showInLineButton,
+    })
   }
-  const fieldId = props.initialValues[props.idField]
-  const modalId = props.isEditable ? `edit-${fieldId || 'new'}-${props.field}-${props.modalId}` : null
 
-  const editButton = props.isEditable && (props.formFields ?
-    <UpdateButton
-      key="edit"
-      modalTitle={props.modalTitle}
-      modalId={modalId}
-      modalSize={props.modalSize}
-      buttonText={props.editLabel}
-      editIconName={props.editIconName}
-      onSubmit={props.onSubmit}
-      initialValues={props.initialValues}
-      formFields={props.formFields}
-      formContainer={<div style={props.modalStyle} />}
-      showErrorPanel={props.showErrorPanel}
-      confirmDialog={props.addConfirm}
-      size="tiny"
-    />
-    : (
-      <DispatchRequestButton
-        key="edit"
-        buttonContent={<Icon link size="small" name="plus" />}
-        onSubmit={() => props.onSubmit(props.initialValues)}
-        confirmDialog={props.addConfirm}
+  render() {
+    if (this.props.isVisible !== undefined && !this.props.isVisible) {
+      return null
+    }
+    if (this.props.isPrivate && !this.props.user.isStaff) {
+      return null
+    }
+    const fieldValue = this.props.initialValues[this.props.field]
+    const hasValue = (fieldValue && (!Object.getOwnPropertyNames(fieldValue).includes('length') || fieldValue.length > 0)) || this.props.showEmptyValues
+    if (!this.props.isEditable && !hasValue) {
+      return null
+    }
+    const fieldId = this.props.initialValues[this.props.idField]
+    const modalId = this.props.isEditable ? `edit-${fieldId || 'new'}-${this.props.field}-${this.props.modalId}` : null
+
+    const updateButton = this.props.showInLine ?
+      <div>
+        {this.state.showInLineButton &&
```

Comment:
rather than doing a conditional for `this.state.showInLineButton` followed immediately by another conditional for `!this.state.showInLineButton` you could just use a ternary:
```
{this.state.showInLineButton ?
  <ButtonLink  icon="write"  onClick={this.toggleButtonVisibility}  /> : 
  <Segment>
          <ReduxFormWrapper
            noModal
            onSubmit={this.props.onSubmit}
            closeInlineModal={this.toggleButtonVisibility}
            form={this.props.modalId}
            initialValues={this.props.initialValues}
            fields={this.props.formFields}
            showErrorPanel={this.props.showErrorPanel}
            confirmDialog={this.props.confirmDialog}
            confirmCloseIfNotSaved
          />
  </Segment>
}
```

Reply:
ESLint: The no-nested-ternary rule disallows nested ternary expressions.